Vingette for step by step guide how to perform t-test. (#286)
Olink-Proteomics/OlinkRPackage: OlinkAnalyze/vignettes/study_design_2groups.Rmd

---
title: "A Guide to Comparative Analysis of Two Groups with Olink Data"
author: "Olink DS team"
date: "`r Sys.Date()`"
output: 
  html_vignette:
    latex_engine: xelatex
    toc: true
    toc_depth: 2
vignette: >
  %\VignetteIndexEntry{A Guide to Comparative Analysis of Two Groups with Olink Data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
--- 

```{r setup, include = FALSE}
knitr::opts_chunk$set(
collapse = TRUE,
comment = "#>")
options(tibble.print_min = 4L, tibble.print_max = 4L)
library(dplyr)
library(stringr)
library(ggplot2)
library(OlinkAnalyze)

df <- OlinkAnalyze::npx_data1 %>% 
  filter(!str_detect(SampleID, 'CONTROL'))
df_QC <- OlinkAnalyze::npx_data1 %>% 
  filter(!str_detect(SampleID, 'CONTROL'))

```

# Introduction  

> This vignette is designed to assist users working with two distinct groups using t-test. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing and comparing data from two groups in Olink data using the Olink R package. These two groups can encompass a variety of categories, such as healthy and diseased individuals, two different types of a condition, or male and female subjects. The objective of the analysis is to identify the differing proteins between the two groups.

Reviewer: ```suggestion
This vignette is designed to assist users working with two distinct groups using t-test. This guide aims to provide you with a comprehensive overview of the steps involved in analyzing and comparing data from two groups in Olink data using Olink's R package, Olink Analyze. These two groups can encompass a variety of categories, such as healthy and diseased individuals, two different types of a condition, or male and female subjects. The objective of the analysis is to identify the differing proteins between the two groups.
```

By following this guide, you will gain a solid understanding of how to effectively compare and contrast two groups, detect differences, and ultimately derive valuable conclusions from your data. Example code will be shown in this vignette, however it is still recommended to read the documentation for each function as not all options will be covered here. This can be done with the _help_ command.

```{r, eval=FALSE}
help("functionName")
```

> If you have more that two groups in your data, or a longitudinal study see our other study design vignettes. 

Reviewer: ```suggestion
If you have more that two groups in your data, or a longitudinal study, see our other study design vignettes. 
```

# Installation 

You can install Olink® Analyze from CRAN which is needed for the following walkthrough.

```{r, eval=FALSE}
install.packages("OlinkAnalyze")
```

After installing the package, the package needs to be loaded. Several packages from the tidyverse suite of packages are also used throughout this vignette, such as dplyr and stringr. Each individual package can be loaded using the _library_ function (`library("package")`) or the entire tidyverse suite of packages can be loaded as shown below.

```{r, eval=FALSE}
library(OlinkAnalyze)
library(tidyverse)
```

# Study overview

> This vignette will help you if you have Olink data and are unsure on how to compare your two groups. The Olink data file shouldn't be tempered with before loading it into R, as that may cause the data to not be recognized and loaded by the `read_NPX` function. Using the read_NPX function will always give you your data in a long format, where each row represent one sample and one protein. To load in your data in R use the following code:

Reviewer: ```suggestion
This vignette will help you if you have Olink data and are unsure on how to compare your two groups. The Olink data file shouldn't be tempered with before loading it into R, as that may cause the data to not be recognized and loaded by the `read_NPX` function. Using the read_NPX function will always give you your data in a long format, where each row represent one sample and one protein. To load in your data in R, use the following code:
```

```{r message=FALSE, eval=FALSE}
data <- read_NPX("~/NPX_file_location.csv")

> # Argument should be the location of your NPX data. Data are usually in parquet, csv, xlsx or zip format depending on the product.

Reviewer: ```suggestion
# Argument should be the location of your NPX data. 
# Data are usually in parquet, csv, xlsx or zip format. 
# The file format will depend on the product.
```
